### ISSUE TITLE
Questions about GTA loss in prop.py

### ISSUE THREAD

**Issue by qzzqzzb** [HarrialX/GraphBackdoor#3]
Hi,

There's some loc I am confused about. 

https://github.com/HarrialX/GraphBackdoor/blob/36e51e876eebb5928d1643ab051cbd12963746fa/trojan/prop.py#L80-L87

It looks like the loss is equal to the sum of losses on positive samples and negative samples. And the cross_entropy between model output and label (data[4]) is sumed up in each loop. But according to the paper, the loss is supposed to be the embedding dissimilarity (equation 3). So why we use this loss function here?

Any insights?

**Comment by zhaohan-xi** [comment]
Hi There, nice catch! Our original implementation follows [this ccs paper](https://dl.acm.org/doi/10.1145/3319535.3354209) who uses an intermediate feature map to optimize. However, we further find it is not always necessary for simpler GNNs because it's not as deep as those vision models, hence we release an end-to-end propagation objective that leverages final labels to optimize, we also posted this change in our README.md -- optimization objectives :)

**Comment by qzzqzzb** [comment]
Thank you for your reply!
